### ISSUE TITLE
Clone FNIRSI replace with ST32F103C8

### ISSUE THREAD

**Issue by driehoek91** [mean00/lnDSO150#8]
mean00 can you compile me the firmware for the fnirsi clone which has been replaced with st32f103c8 so that it has all the features in InDSO150, at first I wanted to compile it myself, but I don't know what configurations to change, thank you...

**Comment by mean00** [comment]
Please read carefully : 
* The attached bin file is for a STM32F103C with FNIRSI button layout and WITH 128 kB of flash (~ all of them do)
* You *NEED* to also flash the bootloader (to check it works power on the device with the OK button pressed ; you should see a usb logo on screen)

absolutely not tested

[f1.zip](https://github.com/mean00/lnDSO150/files/10287699/f1.zip)

**Comment by driehoek91** [comment]
sorry I didn't give the result for a long time, it seems I got a STM32f103C8T6 clone from the 5 bluepills that I bought (high density 0x414 256Kb flash revision Z), so it only shows a white screen, I tried moving the STM32 again from the Fnirsi pcb to the DSO150 pcb and  it only shows a white screen with InDSO150 v2.2, When using DSO_STM32Duino v1.7 it only shows a splash screen then a black screen, but strangely with DSO_STM32Duino v1.6 it can run normally.  I have a feeling maybe this STM32 can't perfectly emulate the USB because I tried to flash it with the ST-Link firmware, but the message USB device is not recognized appears.

**Comment by mean00** [comment]
Just checking : did you flash the lnDSO bootloader first ?

If you boot with "OK" pressed do you see a usb logo ?
If so, do you see a spinning circle ?

**Comment by driehoek91** [comment]
yes I've tried flashing bootloader v2.2 elf/bin with st-link first, and when turning it on while holding down the OK button it just shows a white screen.

> My old STM32 runs perfectly on all versions of InDSO150 and DSO_STM32Duino
![IMG_20230110_025036](https://user-images.githubusercontent.com/56748092/211398466-91b66200-c660-4ae6-9ffa-7abb585b8743.jpg)\

![st ori](https://user-images.githubusercontent.com/56748092/211398482-f98211bb-4563-4d7d-8ba1-422c9701efb3.jpg)

![IMG_20230110_024608](https://user-images.githubusercontent.com/56748092/211398488-0762c381-2e82-407b-bbe4-cc717b1a9ab6.jpg)

![IMG_20230110_030337](https://user-images.githubusercontent.com/56748092/211398524-9d743ee4-70ac-4bf0-9f7c-c23ea93e1c20.jpg)

> My new STM32 failed to run properly, and can only run with DSO_STM32Duino V1.6 and below, other than that it's just a white screen
![IMG_20230110_025001](https://user-images.githubusercontent.com/56748092/211399104-b2edabe3-df88-4a14-9354-4fb576a90e88.jpg)

![st kw](https://user-images.githubusercontent.com/56748092/211399117-22ff706a-4e95-4e16-9373-94d5fac36d7c.jpg)

**Comment by mean00** [comment]
Thanks for the pics, it helps.
I noiced the 2nd says 256kB of flash ? While they are both C8T6 ?
That's strange

**Comment by driehoek91** [comment]
yes, it's true that there is 256kB because I tried to maximize the file that I uploaded, and downloaded it again, the result is the same, it's not corrupt.  I have tried the serial output of InDS0150 and DS0_STM32Duino, but it only writes "main".  and I've also tried gdb with black magic probe, but the result target chips 0x00 and target can't be attached.  I think I give up with this stm32 clone, I'd better buy a bluepill with the original stm32 first

**Comment by mean00** [comment]
I see 2 options :
Size
-------
One difference between stm32duino 1.6 & 1.7 is the former is <64kB the later is not.
If you try the normal lnDSO 2.2 fnirsi version do you have something on the screen ?
That version has 2 flags activated :
- Small footprint so it fits within 64 kB (no font anti alising, no usb, ...)
- Fnirsi PCB changes (button wiring)
![rd](https://user-images.githubusercontent.com/8426909/211474290-671be6c8-fefd-4963-a173-d06eac6a86b1.jpg)


If you have the wrong PCB, it will make the buttons not working but you should see something on screen

Screen ID
---------------
It may be the screen is different / not wired the same and not recognized
Maybe older stm32Dso fws were falling back in a nicer way to ILI9341 while more recent ones
dont work well in that case.
That would be "RD" signal that goes to the screen

**Comment by mean00** [comment]
It might be a CS32F103

**Comment by driehoek91** [comment]
At first I also thought like that, because I have a stock of CKS32 64kB and APM32 128kB.. but when I use CKS32 with option -DUSE_SMALLFOOTPRINT=True, it can run normally. Likewise with APM32 using lnDSO_FNIRSI_2.2.bin release, it works normally like STM32. I think my STM32 clone is really very poor quality

![IMG_20230111_041622](https://user-images.githubusercontent.com/56748092/211665868-da9dd7ab-e2fb-4792-a951-76a41304d893.jpg)

![IMG_20230111_041551](https://user-images.githubusercontent.com/56748092/211665907-2c7c24f2-c52f-422b-b682-9a194118c062.jpg)
